Mathlib/Analysis/Analytic/IsolatedZeros.lean

```diff
@@ -133,6 +133,15 @@ theorem eventually_eq_or_eventually_ne (hf : AnalyticAt 𝕜 f z₀) (hg : Analy
     (∀ᶠ z in 𝓝 z₀, f z = g z) ∨ ∀ᶠ z in 𝓝[≠] z₀, f z ≠ g z := by
   simpa [sub_eq_zero] using (hf.sub hg).eventually_eq_zero_or_eventually_ne_zero
 
+/-- Two analytic functions agree on a punctured neighborhood iff they agree on a neighborhood. -/
+theorem eventuallyEq_nhdNE_iff_eventuallyEq_nhd (hf : AnalyticAt 𝕜 f z₀) (hg : AnalyticAt 𝕜 g z₀) :
+  f =ᶠ[𝓝[≠] z₀] g ↔ f =ᶠ[𝓝 z₀] g := by
```

Comment:
Don't you need only assume one of the functions is analytic?
```suggestion
theorem eventuallyEq_nhdNE_iff_eventuallyEq_nhd (hf : AnalyticAt 𝕜 f z₀) :
  f =ᶠ[𝓝[≠] z₀] g ↔ f =ᶠ[𝓝 z₀] g := by
```

Reply:
I think you need both. For a counterexample, take `f = 0` and take `g` as the indicator function of a point. Then, `f` is analytic and the two agree in a punctured neighbourhood, but in no regular neighbourhood of that point.

Reply:
So you can get away with
```suggestion
theorem eventuallyEq_nhdNE_iff_eventuallyEq_nhd (hf : ContinuousAt 𝕜 f z₀) :
  f =ᶠ[𝓝[≠] z₀] g ↔ f =ᶠ[𝓝 z₀] g := by
```
right? I'm thinking it might be useful as a way to prove analycity of `g` knowing only analycity of `f`

Reply:
I see what you are saying. Going further down that line, it will suffice to assume continuity for `f` as well. Let me see if I can work this out before going to the office.

Reply:
I agree with Yaël, this doesn't seem to be specific to analytic functions.

Reply:
As Yaël's comments point out, the theorem has nothing to do with `AnalyticAt`, but is a statement about functions that are continuous at non-isolated points of a topological space. I have therefore removed the statement from `IsolatedZeros.lean` and added a short private lemma in `NormalFormAt.lean`, where I use it twice.

I made this a private lemma because I found it easier to state a quick lemma for analytic functions, compared to the effort involved in stating/proving/applying a lemma on continuous functions that requires me to argue that points in topological spaces underlying nontrivially normed fields are non-isolated.